packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -254,6 +263,44 @@ export function authorizationServiceBuilder(
         )
       );
     },
+    async removeClientPurpose({
+      clientId,
+      purposeIdToRemove,
+      organizationId,
+      correlationId,
+      logger,
+    }: {
+      clientId: ClientId;
+      purposeIdToRemove: PurposeId;
+      organizationId: TenantId;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<void> {
+      logger.info(
+        `Removing purpose ${purposeIdToRemove} from client ${clientId}`
+      );
+
+      const client = await retrieveClient(clientId, readModelService);
+      assertOrganizationIsClientConsumer(organizationId, client.data);
+
+      retrievePurposeId(client.data, purposeIdToRemove);
```

Comment:
I would call this function `assertPurposeInClientExists` (or something like that) and put it in the validators, what do you think?

Reply:
I am not sure it will be reused. For now, I removed the function and put the check inside the logic.